src/pages/en/guides/content-collections.md

```diff
@@ -0,0 +1,480 @@
+console.log(docsEntries)
+/*
+[
+	{ id: 'en/getting-started.md', slug: 'en/getting-started', data: {...} },
+	{ id: 'en/structure.md', slug: 'en/structure', data: {...} },
+	{ id: 'es/getting-started.md', slug: 'es/getting-started', data: {...} },
+	{ id: 'es/structure.md', slug: 'es/structure', data: {...} },
+	...
+]
+*/
+```
+
+To fetch from a specific directory **within** a collection, Astro recommends `getCollection` with a filter:
+
+```astro
+---
+import { getCollection } from 'astro:content';
+
+const enDocs = await getCollection('docs', ({ id }) => {
+  // Where `id` is 'en/page-1.md' | 'en/page-2.md' | ...
+  return id.startsWith('en');
+});
+---
+```
+
+## Rendering content with `renderEntry`
+
+You may need to render the contents of these Markdown and MDX entries as well. This is especially useful when generating live URLs from your content entries (see [Generating pages from content collections](#generating-pages-from-content-collections)). 
+
+You can retrieve a `Content` component for use in your Astro files with `renderEntry`:
+
+```astro "renderEntry"
+---
+import { renderEntry, getEntry } from 'astro:content';
+
+const announcementPost = await getEntry('announcements', 'welcome.md');
+// Pass a `getEntry` result or `getCollection` array item:
+const { Content } = await renderEntry(announcementPost);
+// Or pass a valid ID and collection name individually:
+const { Content } = await renderEntry({
+  id: announcementPost.id,
+  collection: announcementPost.collection,
+});
+---
+
+<h1>{announcementPost.data.title}</h1>
+<Content />
+```
+
+This example will render the contents of `content/announcements/welcome.md`.
+
+### Access content headings
+
+Astro [generates a list of headings](/en/guides/markdown-content/#exported-properties) for Markdown and MDX documents. You can access this list using the `headings` property from `renderEntry`:
+
+```astro "{ headings }"
+---
+import { getCollection, renderEntry } from 'astro:content';
+const blogPosts = await getCollection('blog');
+---
+
+{blogPosts.map(async (post) => {
+  const { headings } = await renderEntry(post);
+  const h1 = headings.find(h => h.depth === 1);
+  return <p>{h1}</p>
+})}
+```
+
+### Access injected frontmatter
+
+Astro allows you to [inject frontmatter using remark or rehype plugins.](/en/guides/markdown-content/#example-injecting-frontmatter) You can access these values using the `injectedFrontmatter` property from `renderEntry`:
+
+```astro "{ injectedFrontmatter }"
+---
+import { getCollection, renderEntry } from 'astro:content';
+const blogPosts = await getCollection('blog');
+---
+
+{blogPosts.map(async (post) => {
+  const { injectedFrontmatter } = await renderEntry(post);
+  const { readingTime } = injectedFrontmatter;
+  return <p>{post.data.title} - {readingTime}</p>
+})}
+```
+
+Assuming `readingTime` was injected ([see our reading time example](/en/guides/markdown-content/#example-calculate-reading-time)), all properties should be available here.
```

Comment:
```suggestion
Assuming `readingTime` was injected ([see our reading time example](/en/guides/markdown-content/#example-calculate-reading-time)), it will be available on the `injectedFrontmatter` object.
```